cai_causal_graph/graph_components.py

```diff
@@ -23,6 +24,65 @@
 from cai_causal_graph.type_definitions import EDGE_T, NodeLike, NodeVariableType
 
 
+def get_variable_name_and_lag(node_name: NodeLike) -> Tuple[str, int]:
+    """
+    Extract the variable name from a node name.
+
+    Example:
+        'X lag(n=2)' -> 'X' if lagged in the past,
+        'X future(n=2)' -> 'X' if lagged in the future.
+    """
+    # get the string name if the node is a Node object
+    if isinstance(node_name, Node):
+        node_name = node_name.identifier
+    assert isinstance(node_name, str), f'Invalid node name: {node_name}.'
+    match = re.match(r'(\w+) lag\(n=(\d+)\)', node_name)
+
+    match = re.match(r'(\w+)(?: lag\(n=(\d+)\))?(?: future\(n=(\d+)\))?', node_name)
+
+    if match:
+        variable_name = match.group(1)
+        past_lag = match.group(2)
+        future_lag = match.group(3)
+
+        if past_lag:
+            return variable_name, -int(past_lag)
+        elif future_lag:
+            return variable_name, int(future_lag)
+        else:
+            return variable_name, 0  # no lag information
+
+    else:
+        raise ValueError(f'Invalid node name: {node_name}')
+
+
+def get_name_from_lag(variable_name: str, lag: int):
+    """
+    Get the name of a lagged variable.
+    If the lag is 0, then the variable name is returned.
+    If the lag is not 0, then the old lag is removed and
+    variable name is appended with the lag.
+
+    Example:
+        'X', 2 -> 'X lag(n=2)', # lagged in the past
+        'X', -2 -> 'X future(n=2)', # lagged in the future
```

Comment:
Doesn't seem to match code below as negative is past lag and positive is future lag in the code.